sdk/core/AzureCore/Source/Errors.swift

```diff
@@ -64,3 +64,19 @@ public enum AzureError: BaseError {
         }
     }
 }
+
+extension String: LocalizedError {
```

Comment:
I don't recall if we discussed this in the meeting - what is the use case here?

Reply:
This was something I upleveled from Autorest.Swift. @sacheu can you touch on this?

Reply:
This is is to support the proper generation of AzureError when the error response body is something other than a JSON object (i.e. string, int, etc). This was created in Autorest.Swift to support the error-body testserver swagger, but should really live in AzureCore.

Reply:
This is added in order to support xms-error-responses.json autorest testsever test swagger